src/lib/journey/callbacks/_utilities/callback.utilities.ts

```diff
@@ -0,0 +1,374 @@
+  const failedPolicies = callback.getFailedPolicies && callback.getFailedPolicies();
+  const parsedPolicies = parseFailedPolicies(failedPolicies, label);
+
+  return parsedPolicies.map((policy) => {
+    return {
+      params: policy?.params,
+      policyRequirement: policy?.policyRequirement || '',
+      restructured: getValidationFailureParams(policy),
+    };
+  });
+}
+
+export function getValidationPolicies(policies: StringDict<unknown>, label: string): Policy[] {
+  if (typeof policies !== 'object' && !policies) {
+    return [];
+  }
+
+  const reqs = policies?.policies;
+  if (!Array.isArray(reqs)) {
+    return [];
+  }
+  return reqs.map((policy) => {
+    return {
+      message: getValidationMessageString(policy),
+      ...(policy?.params && { params: policy?.params }),
+      ...(policy?.policyId && { policyId: policy?.policyId }),
+    };
+  }).filter((policy) => !!policy.message);
+}
+
+export function isInputRequired(callback: ValidatedCallbacks): boolean {
+  const policies = callback.getPolicies && callback.getPolicies();
+
+  let isRequired = false;
+
+  if (policies?.policyRequirements) {
+    isRequired = policies.policyRequirements.includes('REQUIRED');
+  } else if (callback.isRequired) {
+    isRequired = callback.isRequired();
+  }
+
+  return isRequired;
+}
+
+function convertCharacterSetToRuleObj(set: string) {
+  const arr = set.split(':');
+  const num = arr[0];
+  const type = arr[1];
+
+  if (type === '0123456789') {
+    return {
+      length: Number(num),
+      message: interpolate('minimumNumberOfNumbers', {num: String(num)}),
+      rule: 'numbers',
+    };
+  } else if (type === 'ABCDEFGHIJKLMNOPQRSTUVWXYZ') {
+    return {
+      length: Number(num),
+      message: interpolate('minimumNumberOfUppercase', {num: String(num)}),
+      rule: 'uppercase',
+    };
+  } else if (type === 'abcdefghijklmnopqrstuvwxyz') {
+    return {
+      length: Number(num),
+      message: interpolate('minimumNumberOfLowercase', {num: String(num)}),
+      rule: 'lowercase',
+    };
+  } else if (
+    type.includes('@') ||
+    type.includes('!') ||
+    type.includes('*') ||
+    type.includes('#')
+  ) {
+    return {
+      length: Number(num),
+      message: interpolate('minimumNumberOfSymbols', {num: String(num)}),
+      rule: 'symbols',
+    };
+  } else {
+    return {
+      length: Number(num),
+      message: interpolate('pleaseCheckValue'),
+      rule: 'unknown',
+    };
+  }
+}
+
+export function parseFailedPolicies(
+  policies: unknown[],
+  label: string,
+): (PolicyRequirement | undefined)[] {
+  return policies.map((policy) => {
+    if (typeof policy === 'string') {
+      try {
+        return JSON.parse(policy) as PolicyRequirement;
+      } catch (err) {
+        console.log(`Parsing failure for ${label}`);
+      }
+    } else {
+      return policy as PolicyRequirement;
+    }
+  });
+}
+
+/** *********************************************
+ * OLD METHODS
```

Comment:
The below are old methods that we need to remove once we can validate we don't need them anymore. The "new" methods from above should cover all use cases we need.